packages/json/src/browser/json-validation-registry.ts

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (c) 2020 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from 'inversify';
+import { Disposable } from '@theia/core/lib/common/disposable';
+import { deepClone } from '@theia/core';
+import { JsonSchemaConfiguration } from '@theia/core/lib/browser/json-schema-store';
+
+@injectable()
+/**
+ * Holds JSON Validation contributions from plugins / VSCode extensions.
+ * -  https://code.visualstudio.com/api/references/contribution-points#contributes.jsonValidation
+ */
+export class JsonValidationContributionsRegistry {
+
+    protected readonly registry: JsonSchemaConfiguration[] = [];
+
+    registerJsonValidation(jsonValidationEntry: JsonSchemaConfiguration): Disposable {
```

Comment:
Why do we need it? There is already `JsonSchemaStore.registerSchema`, please use it.

Reply:
The reason for a separate registry is to allow applying **precedence** to the schema depending on its "origin".

See: https://github.com/eclipse-theia/theia/pull/7560/files#diff-31a8b9aa2aa16c406d15ae82f1fdc3eaR54-R61

If I understand correctly this is already happening in regards to schemas originating from the schema store as those originating from user preferences.

See: https://github.com/eclipse-theia/theia/blob/master/packages/json/src/browser/json-client-contribution.ts#L52-L56

Basically the problem is that given two schemas configs with **identical** `fileMatch` property, only the **last** one will be applied and we need a manner to define which take precedence.

Also see this comment on how it should work in VSCode:
- https://github.com/microsoft/vscode/issues/26289#issuecomment-578768254